src/NzbDrone.Core/Datastore/Migration/023_add_release_groups_etc.cs

```diff
@@ -0,0 +1,105 @@
+                .WithColumn("Type").AsString()
+                .WithColumn("SecondaryTypes").AsString().Nullable()
+                .WithColumn("Ratings").AsString().Nullable()
+                .WithColumn("CleanTitle").AsString().Indexed()
+                .WithColumn("ProfileId").AsInt32().Nullable()
+                .WithColumn("Monitored").AsBoolean()
+                .WithColumn("LastInfoSync").AsDateTime().Nullable()
+                .WithColumn("LastDiskSync").AsDateTime().Nullable()
+                .WithColumn("Added").AsDateTime().Nullable()
+                .WithColumn("AddOptions").AsString().Nullable()                
+                .WithColumn("SelectedReleaseId").AsInt32();
+
+            Create.TableForModel("Release")
+                .WithColumn("ForeignReleaseId").AsString().Unique()
+                .WithColumn("ReleaseGroupId").AsInt32().Indexed()
+                .WithColumn("Title").AsString()
+                .WithColumn("Status").AsString()
+                .WithColumn("Duration").AsInt32().WithDefaultValue(0)
+                .WithColumn("Label").AsString().Nullable()
+                .WithColumn("Disambiguation").AsString().Nullable()
+                .WithColumn("Country").AsString().Nullable()
+                .WithColumn("Media").AsString().Nullable()
+                .WithColumn("TrackCount").AsInt32().Nullable();
+
+            Create.TableForModel("Track")
+                .WithColumn("ForeignTrackId").AsString().Unique()
+                .WithColumn("ArtistMetadataId").AsInt32().Indexed()
+                .WithColumn("ReleaseId").AsInt32().Indexed()
+                .WithColumn("TrackNumber").AsString().Nullable()
+                .WithColumn("AbsoluteTrackNumber").AsInt32()
+                .WithColumn("MediumNumber").AsInt32().Nullable()
+                .WithColumn("Title").AsString().Nullable()
+                .WithColumn("Duration").AsInt32().WithDefaultValue(0)
+                .WithColumn("Explicit").AsBoolean()
+                .WithColumn("Ratings").AsString().Nullable()
+                .WithColumn("TrackFileId").AsInt32().Indexed();
+
+            Create.TableForModel("TrackFile")
+                .WithColumn("RelativePath").AsString().Nullable()
+                .WithColumn("Size").AsInt64()
+                .WithColumn("DateAdded").AsDateTime()
+                .WithColumn("SceneName").AsString().Nullable()
+                .WithColumn("ReleaseGroup").AsString().Nullable()                
+                .WithColumn("Quality").AsString()
+                .WithColumn("MediaInfo").AsString().Nullable()
+                .WithColumn("Language").AsInt32().WithDefaultValue(0)
+                .WithColumn("AlbumId").AsInt32();
+
+            // TODO: Actually migrate the data
+
+            // Drop the old Artists, Albums, Tracks, TrackFiles tables
+            Delete.Table("Artists");
+            Delete.Table("Albums");
+            Delete.Table("Tracks");
+            Delete.Table("TrackFiles");
```

Comment:
Yeah, now I've started doing the actual migration I've done it how you suggest - editing the existing tables.

At the moment there is still ReleaseGroup (which has extra information) and Album. I've done the minimal renaming to make the new schema work, so mostly on adding new items where the extra info is required. ReleaseGroup is then implicitly cast to Album in the rest of the code to avoid messing with it. Happy to try a different approach if you think it would be better. I guess I could try augmenting the album class rather than creating a new one but was worried I'd get myself in a mess around when the extra info is populated or not

Reply:
Thinking about this some more, I think you're probably right that I could merge the ReleaseGroup class into Album, retaining the old properties by calculating from the new ones. That would probably be easier than casting to Album. I'll have a go.

Reply:
Sounds good, we could probably do this in stages to if you want to break it up. Something like:

- break out releases into table
- add artist metadata table
- etc.

Up to you on that though